compiler/rustc_hir_typeck/src/op.rs

```diff
@@ -974,8 +975,14 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
             },
         );
 
-        let method =
-            self.lookup_method_for_operator(cause.clone(), opname, trait_did, lhs_ty, opt_rhs_ty);
+        let method = self.lookup_method_for_operator(
+            cause.clone(),
+            opname,
+            trait_did,
+            lhs_ty,
+            opt_rhs_ty,
+            TreatNotYetDefinedOpaques::AsInfer,
```

Comment:
feel like we should have some justification for why we don't treat them as rigid here. I think the answer is just that we know syntactically what operator it is and don't need to do trait solving to "resolve" anything?